### PR TITLE
Immersive Portals - Client crash fix

### PR DESCRIPTION
**Pull request by Jeryn99**
Fixes a issue where the game tries to attach a renderer to a entity that doesn't exist

### CODE REVIEW

File: common/src/main/java/whocraft/tardis_refined/common/crafting/astral_manipulator/ManipulatorCraftingRecipe.java [hunk 1]
```diff
@@ -127,6 +127,14 @@ public boolean canCraftInDimensions(int width, int height) {
 
     @Override
     public ItemStack getResultItem(RegistryAccess registryAccess) {
+        if(result.type() instanceof ManipulatorItemResult manipulatorItemResult){
+            return manipulatorItemResult.recipeOutput();
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/common/crafting/astral_manipulator/ManipulatorCraftingRecipe.java:131]: Why is this needed? I thought this only get used in vanilla crafting benches.

File: common/src/main/java/whocraft/tardis_refined/compat/portals/BOTIPortalEntity.java [hunk 2]
```diff
@@ -79,4 +104,11 @@ public void setTardisId(UUID tardisId) {
         this.getEntityData().set(TARDIS_ID, Optional.of(tardisId));
     }
 
+    public boolean isValid() {
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/compat/portals/BOTIPortalEntity.java:101]: Javadocs for what this does would be nice.

**Review by 50ap5ud5 — CHANGES_REQUESTED** [review]
Looking good, some minor documentation needed

File: common/src/main/java/whocraft/tardis_refined/common/crafting/astral_manipulator/ManipulatorCraftingRecipe.java (hunk 1 above)
Comment (Jeryn99) [reply common/src/main/java/whocraft/tardis_refined/common/crafting/astral_manipulator/ManipulatorCraftingRecipe.java:131]: You're correct, but if for some weird reason a mod tries to render our recipes out, its better to render something than nothing

Comment (50ap5ud5) [reply common/src/main/java/whocraft/tardis_refined/common/crafting/astral_manipulator/ManipulatorCraftingRecipe.java:131]: Good catch, nice one.

File: common/src/main/java/whocraft/tardis_refined/compat/portals/ImmersivePortals.java [hunk 3]
```diff
@@ -351,16 +355,29 @@ public static void destroyPortals(TardisLevelOperator operator) {
         if (portalEntry == null) {
             return;
         }
+
+        PortalManipulation.removeConnectedPortals(portalEntry.getInternalPortal(), portal -> {
+
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/compat/portals/ImmersivePortals.java:360]: Is this meant to be left empty?

Comment (Jeryn99) [reply common/src/main/java/whocraft/tardis_refined/compat/portals/ImmersivePortals.java:360]: Yes

**Review by 50ap5ud5 — APPROVED** [review]
LGTM
